packages/teams-js/src/internal/communication.ts

```diff
@@ -315,18 +386,16 @@ const shouldProcessMessageLogger = communicationLogger.extend('shouldProcessMess
 function shouldProcessMessage(messageSource: Window, messageOrigin: string): boolean {
   // Process if message source is a different window and if origin is either in
   // Teams' pre-known whitelist or supplied as valid origin by user during initialization
-  if (Communication.currentWindow && messageSource === Communication.currentWindow) {
+  if (isMessageComingFromCurrentWindow(messageSource)) {
     shouldProcessMessageLogger('Should not process message because it is coming from the current window');
     return false;
-  } else if (
-    Communication.currentWindow &&
-    Communication.currentWindow.location &&
-    messageOrigin &&
-    messageOrigin === Communication.currentWindow.location.origin
-  ) {
+  } else if (isMessageComingFromSelf(messageOrigin)) {
```

Comment:
I believe this check is in the case of a dialog or a child window, I could change the name of the function to represent that, but if this if were to come first it would effectively pass whenever this app sends out a message to itself. That's the only reason why I think this check is in place.

Reply:
I believe Jay's correct, this is really checking for "message from my own domain" (which is sort of what "fromSelf" means, but maybe a little less ambiguous. I *think* Jay's scenarios are also the ones I assumed this was for (dialogs that want to use the back channel for communicating with their parent app for example)